pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -56,45 +62,65 @@ type LaunchTemplateProvider struct {
 	clientSet             *kubernetes.Clientset
 }
 
-func launchTemplateName(clusterName string, arch string) string {
-	return fmt.Sprintf(launchTemplateNameFormat, clusterName, arch)
+func launchTemplateName(options *launchTemplateOptions) string {
+	hash, err := hashstructure.Hash(options, hashstructure.FormatV2, nil)
+	if err != nil {
+		zap.S().Panicf("hashing launch template, %w", err)
+	}
+	return fmt.Sprintf(launchTemplateNameFormat, options.Cluster.Name, options.Provisioner.Name, options.Provisioner.Namespace, fmt.Sprint(hash))
+}
+
+type launchTemplateOptions struct {
```

Comment:
maybe add a comment here that this struct *must not* contain any fields that change trivially (otherwise we'll end up with "leaked" launch templates)

Reply:
Good point!